crates/bevy_render/src/view/window.rs

```diff
@@ -128,6 +138,11 @@ fn extract_windows(
     for closed_window in closed.iter() {
         extracted_windows.remove(&closed_window.id);
     }
+    for (window, screenshot_func) in screenshot_manager.callbacks.lock().drain() {
```

Comment:
Probably should use `try_lock` here to avoid a deadlock if a screenshot is being taken at the same time? Although I guess the bevy scheduler prevents that from happening here..

Reply:
Yep, this can't really happen because `callbacks` is `pub(crate)` and this is the singular callsite where it's called. Even if somehow a user ended up with two of these systems, one would simply take the lock and they'd run in series. Only one lock isn't enough to cause a deadlock.

Reply:
Nile, your comment here would make a great comment for this line of code.